spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/MultiMethodRabbitListenerEndpoint.java

```diff
@@ -47,7 +52,36 @@ protected HandlerAdapter configureListenerAdapter(MessagingMessageListenerAdapte
 			invocableHandlerMethods.add(getMessageHandlerMethodFactory()
 					.createInvocableHandlerMethod(getBean(), method));
```

Comment:
I wonder if we could end up with some issue, having `bean` as a field here and the setter for the same in the super class...
I'd say that we really should use for the `MultiMethodRabbitListenerEndpoint` the same setters model. Even for the `methods`.